Assets/Scripts/StaminaBar.cs

```diff
@@ -46,26 +42,46 @@ public void recover()
         // Standard recovery
         if (stamina < maxStamina && !crash)
         {
-            stamina += standardRestore;
-            Debug.Log(stamina);
+            if (debug)
+                Debug.Log("Standard Restore: ");
+            delta(standardRestore);
         }
         // Crash Debuff recovery
         else if (stamina < maxStamina && crash)
         {
-            stamina += crashRestore;
-            Debug.Log(stamina);
+            if (debug)
+                Debug.Log("Crashed Restore: ");
+            delta(crashRestore);
         }
-
-        if (stamina == maxStamina)
-            crash = false;
     }
 
     // Drains stamina by n when called.
     public void attack()
     {
-        Debug.Log("Attacked: " + stamina);
+        if(debug)
+            Debug.Log("Attacked: ");
+        delta(-attackDrain);
+    }
 
-        if (stamina > 0)
-            stamina -= attackDrain;
+    // Changes stamina by delta
+    public void delta(int delta)
+    {
+        // Alter stamina
+        stamina += delta;
+        if (debug)
+            Debug.Log(stamina);

```

Comment:
Please remove the debug messages before making a pull request. Its okay for play testing but its going to clog the console for everyone else